boa/src/syntax/ast/node/operator/bin_op/mod.rs

```diff
@@ -86,7 +87,12 @@ impl Executable for BinOp {
                     NumOp::Sub => x.sub(&y, interpreter),
                     NumOp::Mul => x.mul(&y, interpreter),
                     NumOp::Exp => x.pow(&y, interpreter),
-                    NumOp::Div => x.div(&y, interpreter),
+                    NumOp::Div => {
+                        if y == Value::bigint(BigInt::from(0)) {
+                            return interpreter.throw_range_error("BigInt division by zero");
+                        }
+                        x.div(&y, interpreter)
+                    }
```

Comment:
we should change the behaviour of `Value::div()` not the the `BinOp` one, it should not be changed here it should just call `Value::div()`